File: src-tools/get-repo-data.js
/* eslint-disable no-console, no-await-in-loop */
const fs = require('fs');
const { Octokit } = require('@octokit/rest');

const gh = new Octokit({
  version: '3.0.0',
  debug: true,
  headers: { 'user-agent': 'akx.github.io-buildtool' },
});

function complete(repos) {
  fs.writeFileSync('repos.json', JSON.stringify(repos), 'UTF-8');
}

async function getContent() {
  let repos = [];
  for (let page = 1; ; page += 1) {
    console.log('...', page);
    const res = await gh.repos.listForUser({
      username: 'akx',
      page,
      per_page: 100,
    });
    repos = repos.concat(res.data);
    if (res.data.length <= 0) {
      break;
    }
  }
  return repos;
}

getContent().then(complete);

File: src-tools/get-repo-data.py
# /// script
# dependencies = ["httpx"]
# ///
import httpx
import json
from itertools import count


def get_content():
    with httpx.Client() as c:
        for page in count(1):
            print(f"... {page}")
            response = c.get(
                "https://api.github.com/users/akx/repos",
                params={"page": page, "per_page": 100},
                headers={"User-Agent": "akx.github.io-buildtool"},
            )
            response.raise_for_status()
            data = response.json()
            yield from data
            if not data:
                break

if __name__ == "__main__":
    with open("repos.json", "w", encoding="utf-8") as f:
        json.dump(list(get_content()), f)
